File: examples/getting-started/pulumi-and-structurizr/systems/shop/containers/productDatabase.ts
import { Container } from 'structurizr-typescript';
import { Shop } from '../shop';
import * as azure from '@pulumi/azure';

export class ProductDatabase {

    public container: Container;
    public infrastructure: {
        account: azure.storage.Account,
        products: azure.storage.Table,
        payments: azure.storage.Table
    };

    constructor(shop: Shop) {
        // Define architecture
        this.container = shop.system.addContainer('Database', 'stores available products and customer orders', 'SQL Database')!;

        // Define infrastructure
        const account = new azure.storage.Account('product', {
            resourceGroupName: shop.resourceGroup.name,
            accountTier: 'Standard',
            accountReplicationType: 'LRS'
        })

        const products = new azure.storage.Table('products', {
            resourceGroupName: shop.resourceGroup.name,
            name: 'products',
            storageAccountName: account.name
        });

        const payments = new azure.storage.Table('payments', {
            resourceGroupName: shop.resourceGroup.name,
            name: 'payments',
            storageAccountName: account.name
        });

        this.infrastructure = { account, products, payments };
    }
}

File: labs/dotnet/pulumi-and-structurizr/Systems/Shop/Containers/ProductDatabase.cs
using Pulumi.Azure.Storage;

internal class ProductDatabase
{
    private readonly Shop shop;

    public Structurizr.Container Container { get; private set; }
    public (Account Account, Table Products, Table Payments) Infrastructure { get; private set; }

    public ProductDatabase(Shop shop)
    {
        Container = shop.System.AddContainer("Database", "stores available products and customer orders", "SQL Database");
        this.shop = shop;
    }

    public void DefineInfrastructure()
    {
        var account = new Account("product", new AccountArgs
        {
            ResourceGroupName = shop.ResourceGroup.Name,
            AccountTier = "Standard",
            AccountReplicationType = "LRS"
        });

        var products = new Table("products", new TableArgs
        {
            Name = "products",
            StorageAccountName = account.Name
        });


        var payments = new Table("payments", new TableArgs
        {
            Name = "payments",
            StorageAccountName = account.Name
        });

        Infrastructure = (account, products, payments);
    }
}
